packages/fast-components-react-base/src/radio/radio.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import Foundation, { HandledProps } from "../foundation";
+import { IRadioHandledProps, IRadioManagedClasses, IRadioUnhandledProps, RadioProps, RadioTags } from "./radio.props";
+import { IManagedClasses, IRadioClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * Radio state interface
+ */
+export interface IRadioState {
+    checked: boolean;
+}
+
+class Radio extends Foundation<IRadioHandledProps & IRadioManagedClasses, IRadioUnhandledProps, IRadioState> {
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(nextProps: IRadioHandledProps, prevState: IRadioState): null | Partial<IRadioState> {
+        if (typeof nextProps.checked === "boolean" && nextProps.checked !== prevState.checked) {
+            return {
+                checked: nextProps.checked
+            };
+        }
+
+        return null;
+    }
+
+    /**
+     * Handled props instantiation
+     */
+    protected handledProps: HandledProps<IRadioHandledProps & IManagedClasses<IRadioClassNameContract>> = {
+        checked: void 0,
+        disabled: void 0,
+        managedClasses: void 0,
+        onChange: void 0,
+        tag: void 0,
+        children: void 0
+    };
+
+    /**
+     * Provides reference to input
+     */
+    private inputRef: React.RefObject<HTMLInputElement>;
+
+    /**
+     * Define constructor
+     */
+    constructor(props: RadioProps) {
+        super(props);
+
+        this.state = {
+            checked: this.props.checked || false
+        };
+
+        this.inputRef = React.createRef();
+    }
+
+    public render(): React.ReactElement<HTMLElement> {
+        return (
+            <this.tag
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+            >
+                <input
+                    className={get(this.props, "managedClasses.radio_input")}
+                    type="radio"
+                    ref={this.inputRef}
+                    onChange={this.handleRadioChange}
+                    disabled={this.props.disabled || null}
+                    checked={this.state.checked}
+                />
+                <span className={get(this.props, "managedClasses.radio_stateIndicator")} />
```

Comment:
radio requires a label, so this should always create a label element - I think the best structure for this would be as a sibling to the input element. Then we can make the wrapping element a div and remove the "tag" prop

Reply:
add "slot label"